Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -851,116 +844,71 @@ public override int GetHashCode()
-                            }
-                            catch (Exception e)
-                            {
-                                throw new MzLibUtil.MzLibException(
-                                    "Error while trying to parse string into peptide: " + e.Message);
-                            }
-
-                            if (!idToMod.TryGetValue(modId, out Modification mod))
-                            {
-                                throw new MzLibUtil.MzLibException(
-                                    "Could not find modification while reading string: " + FullSequence);
-                            }
-
-                            if (mod.LocationRestriction.Contains("C-terminal.") && r == FullSequence.Length - 1)
-                            {
-                                currentModificationLocation = baseSequenceSb.Length + 2;
-                            }
-
-                            AllModsOneIsNterminus.Add(currentModificationLocation, mod);
-                            currentlyReadingMod = false;
-                            currentModification = new StringBuilder();
+                            throw new MzLibUtil.MzLibException(
+                                "Error while trying to parse string into peptide: " + e.Message);
                         }
-                        else
+                        if (!idToMod.TryGetValue(modId, out Modification mod))
                         {
-                            currentModification.Append(c);
+                            throw new MzLibUtil.MzLibException(
+                                "Could not find modification while reading string: " + FullSequence);
                         }
-
-                        break;
-
-                    default:
-                        if (currentlyReadingMod)
-                        {
-                            currentModification.Append(c);
-                        }
-                        else
+                        if (mod.LocationRestriction.Contains("C-terminal.") && r == FullSequence.Length - 1)
                         {
-                            currentModificationLocation++;
-                            baseSequenceSb.Append(c);
+                            currentModificationLocation = BaseSequence.Length + 2;
                         }
-                        break;
+                        _allModsOneIsNterminus.Add(currentModificationLocation, mod);
+                        currentlyReadingMod = false;
+                    }
                 }
+                else if (!currentlyReadingMod)
+                {
+                    currentModificationLocation++;
+                }
+                //else do nothing
             }
-
-            baseSequence = baseSequenceSb.ToString();
-        }
-
-        private void GetProteinAfterDeserialization(Dictionary<string, Protein> idToProtein)
```

Comment:
does removing this method really save a lot of time? the exception message thrown in the deleted code is more informative than a generic keynotfound exception

Reply:
We can keep this method. It only added a few seconds.